pagopa/io-app
# fix: [IA-262] Request payment delete when it fails

## Short description
This PR requests a payment delete when the paymenent fails for some reasons

Previously the app asks for a deletion in these cases:

| present or added |                                                                                        when                                                                                       | implicit/explicit |
|:---------------:|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|-------------------|
|        ✅        | user wants to go back while a payment  ![Schermata 2021-10-04 alle 11 59 40](https://user-images.githubusercontent.com/822471/135831938-eb9040ea-168a-4250-ab4b-b408e69adf4b.png) |      explicit     |
|        ✅        | app crashes or user forces it to close while a payment                                                                                                                            |      implicit     |
|        ➕       | the payment fails for some reason                                                                                                                                                 |      implicit     |

### how to test
- change dev-server to response with an error on payment. Change [this code](https://github.com/pagopa/io-dev-api-server/blob/428e59582ee8ef40dccd5b9e54d9b8de3ae1b6c4/src/routers/payment.ts#L157) with an int > 0 
- run a payment
- check the network activities: `wallet/v1/payments` should be requested

more info: https://pagopa.atlassian.net/wiki/spaces/IOAPP/pages/425066747/Cancellare+un+pagamento
